index.js

```diff
@@ -135,11 +148,91 @@ const sendMessage = async (data, topic) => {
         });
         logger.info(`successfully published msg id ${job}:${buildConfig.buildId} -> topic ${topic}`);
     } catch (e) {
-        logger.error(`Publishing message ${buildConfig.buildId} to topic ${topic} failed ${e.message}`);
+        logger.error(
+            `Publishing message ${buildConfig.buildId} to topic ${topic} failed ${e.message}: stack: ${e.stack}`
+        );
     }
 };
 
+/**
+ * Connects to a Kafka client instance as admin
+ * @returns Admin promise object
+ */
+const connectAdmin = async () => {
+    const kafka = await getKafkaObject();
+
+    const admin = kafka.admin();
+    const { DISCONNECT, CONNECT } = admin.events;
```

Comment:
Majority of the code in this function is very similar to [connect](https://github.com/screwdriver-cd/aws-producer-service/pull/6/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R91) function 
Can we abstract duplicate code into to a common function?